### ISSUE TITLE
Route history forgets start points and other details

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#19200]
Not sure if this is by design, but it looks,weird: If your last route calculation was not a Navigation but just a From/To route (with a start point selected other than your own position), on the next (subsequent) call to the route configuration screen that route will be remembered correctly, i.e. by tapping the entry under "Prevois route" you can recall that  complete with its "From" start point and the selected means of transport.

But subsequently this route would get pushed further down to the "History" section, and at that point much (perhaps all) of its details become forgotten, "History" seems to merely remember the destination point? Surely this is not as it should be, why isn't there a fully functional route history?

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced. Testing two routes, Katowice to Tychy and Katowice to Mikolow, reveals that only the last calculation restores the full route from start to finish in the history. If the last route is Katowice to Tychy, it resumes correctly. However, creating a new route like Katowice to Mikolow afterward and trying to restore Katowice to Tychy only restores the destination from history, not the entire route.

OsmAnd~ 4.7.0#1698m, released: 2024-02-28


https://github.com/osmandapp/OsmAnd/assets/127092082/1c431694-5d7d-4c0e-bb92-db5e2a763e3c

**Comment by vshcherb** [comment]
Yes by design. As we don't store start-end-intermediate points and only target points. Since it was implemented it's first complain so far and I think refactoring is very cumbersome to store all route settings and not that important
